bin/agent-data-plane/src/main.rs

```diff
@@ -65,16 +70,31 @@ async fn main() {
         fatal_and_exit(format!("failed to initialize TLS: {}", e));
     }
 
-    match run(started).await {
-        Ok(()) => info!("Agent Data Plane stopped."),
-        Err(e) => {
-            error!("{:?}", e);
-            std::process::exit(1);
+    match cli.action {
```

Comment:
I think we can reduce some of this duplicated code by defining `let run_config: RunConfig` outside the `match` and then just assigning to it.